lib/muchos/config/decorators.py

```diff
@@ -0,0 +1,98 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from collections.abc import Iterable
+from functools import wraps
+
+
+_host_vars = []
+_play_vars = []
+_extra_vars = []
+_ansible_vars = dict(
+    host=[],
```

Comment:
yes I will add documentation.  The middle term is the class name, so that we can filter out any cluster types that don't apply.  Since the decorators are run when definitions (python files) are read, and config/__init__.py is importing all cluster types, all cluster type config definitions are read, and we needed to ensure we weren't trying to render ec2 variables when dealing with azure clusters, and vice versa.  I will make this clear in documentation.

Reply:
yes it is done here https://github.com/apache/fluo-muchos/pull/288/files#diff-42038982bf1b118ed041b8288e516336R150-R153

it's probably not terribly obvious, I will add comments to make it clear...

Reply:
The comments and more structured code are nice.